packages/datadog-plugin-fs/test/index.spec.js

```diff
@@ -0,0 +1,1780 @@
+'use strict'
+
+const agent = require('../../dd-trace/test/plugins/agent')
+const { expectSomeSpan } = require('../../dd-trace/test/plugins/helpers')
+const plugin = require('../src')
+
+const realFS = Object.assign({}, require('fs'))
+const os = require('os')
+const path = require('path')
+const semver = require('semver')
+const rimraf = require('rimraf')
+
+const implicitFlag = semver.satisfies(process.versions.node, '>=11.1.0')
+const hasWritev = semver.satisfies(process.versions.node, '>=12.9.0')
+const hasOSymlink = realFS.constants.O_SYMLINK
+
+wrapIt()
+
+// TODO remove skips
+
+describe('Plugin', () => {
+  describe('fs', () => {
+    let fs
+    let tmpdir
+    let tracer
+    afterEach(() => agent.close())
+    beforeEach(() => agent.load(plugin, 'fs').then(() => {
+      tracer = require('../../dd-trace')
+      fs = require('fs')
+    }))
+    before(() => {
+      tmpdir = realFS.mkdtempSync(path.join(os.tmpdir(), 'dd-trace-js-test'))
+    })
+    after((done) => {
+      rimraf(tmpdir, realFS, done)
+    })
+
+    // For almost every test, we're not testing the case where there's no
+    // parent, so we'll wrap `it` so that it always adds a parent scope.
+    const realIt = global.it
+    function wrappedIt (testFn, name, fn) {
+      function inner (done) {
+        const parentSpan = tracer.startSpan('parent')
+        try {
+          tracer.scope().activate(parentSpan, () => fn(done))
```

Comment:
> Is this really enough duplicated code to justify hiding it in a potentially unexpected way?

It was an attempt to not change the entire file yet again. An alternative approach might be to have a function that just wraps the test function, like:

```js
it('should be instrumented', withParentSpan((done) => {
  /* ... */
}))
```

How about that?

> Another thing worth noting is that it's possible that this would just work by activating in beforeEach if wrapIt is not called at the beginning of the file.

I'll give this a shot.

Reply:
I gave it a shot. As you suggested, by not using `wrapIt`, there's no need to do this kind of thing, so everything's much simpler.